### DISCUSSION TITLE
Make content of diagnostic picker dynamic

### DISCUSSION THREAD

**Discussion by bluthej** [helix-editor/helix#8484] [Ideas]
When working on large projects it takes a bit of time for the language server to serve the diagnostics (usually for me it's big Rust projects with RA), and if I open the diagnostic picker before it's done I'll get an empty list. When the server is done, the picker remains empty, and I need to close and reopen it to get the list of diagnostics.

It would be really cool if I could leave the picker open and it would fill up as the diagnostics are served.

**Comment by the-mikedavis** [comment]
Diagnostics are published asynchronously and `<space>d` brings up a picker that has the ones we've been sent by the picker. It's not a request/response like the symbol picker for example, so we can't use a dynamic picker like we have with the workspace symbols picker. We could stream in new diagnostics if the currently open picker is a diagnostics picker now that we've switched to nucleo (which supports streaming in new items). There isn't a way to indicate that we're waiting on diagnostics though since that information isn't sent by the server.

**Comment by Rudxain** [comment]
Referencing #12667 because GH doesn't link issues with discussions
